sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/utils/ResourceParser.java

```diff
@@ -6,18 +6,32 @@
 import com.azure.monitor.opentelemetry.exporter.implementation.ResourceAttributes;
 import com.azure.monitor.opentelemetry.exporter.implementation.builders.AbstractTelemetryBuilder;
 import com.azure.monitor.opentelemetry.exporter.implementation.models.ContextTagKeys;
-import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
 import io.opentelemetry.sdk.resources.Resource;
 
+import java.util.Locale;
 import java.util.Map;
 
 public final class ResourceParser {
 
     private static final String DEFAULT_SERVICE_NAME = "unknown_service:java";
 
+    // these are only used in App Services environment
+    private final String websiteSiteName;
+    private final String websiteSiteInstance;
+
+    public ResourceParser() {
+        this(System.getenv());
+    }
+
+    // visible for testing
+    ResourceParser(Map<String, String> envVars) {
+        websiteSiteName = getWebsiteSiteNameEnvVar(envVars);
+        websiteSiteInstance = envVars.get("WEBSITE_INSTANCE_ID");
+    }
+
     // visible for testing
-    public static void updateRoleNameAndInstance(
-        AbstractTelemetryBuilder builder, Resource resource, ConfigProperties configProperties) {
+    public void setRoleNameAndInstance(
```

Comment:
simplified this by removing conditional update. I don't think Java agent should use this code anyways, since it already handles app services env vars and HostName.get() during startup. We can call (only) `AksResourceAttributes` directly from Java agent instead.